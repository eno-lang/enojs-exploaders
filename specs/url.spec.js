const eno = require('enojs');
const loaders = require('../index.js');

describe('url', () => {

  let document;

  describe('http://valid.com', () => {
    it('returns the url', () => {
      document = eno.parse('url: http://valid.com');
      expect(document.field('url', loaders.url)).toBe('http://valid.com');
    });
  });

  describe('http://www.valid.com', () => {
    it('returns the url', () => {
      document = eno.parse('url: http://www.valid.com');
      expect(document.field('url', loaders.url)).toBe('http://www.valid.com');
    });
  });

  describe('https://valid.com', () => {
    it('returns the url', () => {
      document = eno.parse('url: https://valid.com');
      expect(document.field('url', loaders.url)).toBe('https://valid.com');
    });
  });

  describe('https://www.valid.com', () => {
    it('returns the url', () => {
      document = eno.parse('url: https://www.valid.com');
      expect(document.field('url', loaders.url)).toBe('https://www.valid.com');
    });
  });

  describe('invalid', () => {
    it('throws an error', () => {
      document = eno.parse('url: invalid');
      expect(() => document.field('url', loaders.url)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('www.invalid', () => {
    it('throws an error', () => {
      document = eno.parse('url: www.invalid');
      expect(() => document.field('url', loaders.url)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('www.invalid.com', () => {
    it('throws an error', () => {
      document = eno.parse('url: www.invalid.com');
      expect(() => document.field('url', loaders.url)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('htp://www.invalid.com', () => {
    it('throws an error', () => {
      document = eno.parse('url: htp://www.invalid.com');
      expect(() => document.field('url', loaders.url)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('http:/invalid.com', () => {
    it('throws an error', () => {
      document = eno.parse('url: http:/invalid.com');
      expect(() => document.field('url', loaders.url)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('https//invalid.com', () => {
    it('throws an error', () => {
      document = eno.parse('url: https//invalid.com');
      expect(() => document.field('url', loaders.url)).toThrowErrorMatchingSnapshot();
    });
  });

  describe('https://invalid', () => {
    it('throws an error', () => {
      document = eno.parse('url: https://invalid');
      expect(() => document.field('url', loaders.url)).toThrowErrorMatchingSnapshot();
    });
  });
});
