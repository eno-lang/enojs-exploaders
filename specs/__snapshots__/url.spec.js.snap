// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`url htp://www.invalid.com throws an error 1`] = `
"'url' must be a url (e.g. should look like http(s)://example.com).

   Line | Content
 >    1 | url: htp://www.invalid.com
"
`;

exports[`url http:/invalid.com throws an error 1`] = `
"'url' must be a url (e.g. should look like http(s)://example.com).

   Line | Content
 >    1 | url: http:/invalid.com
"
`;

exports[`url https//invalid.com throws an error 1`] = `
"'url' must be a url (e.g. should look like http(s)://example.com).

   Line | Content
 >    1 | url: https//invalid.com
"
`;

exports[`url https://invalid throws an error 1`] = `
"'url' must be a url (e.g. should look like http(s)://example.com).

   Line | Content
 >    1 | url: https://invalid
"
`;

exports[`url invalid throws an error 1`] = `
"'url' must be a url (e.g. should look like http(s)://example.com).

   Line | Content
 >    1 | url: invalid
"
`;

exports[`url www.invalid throws an error 1`] = `
"'url' must be a url (e.g. should look like http(s)://example.com).

   Line | Content
 >    1 | url: www.invalid
"
`;

exports[`url www.invalid.com throws an error 1`] = `
"'url' must be a url (e.g. should look like http(s)://example.com).

   Line | Content
 >    1 | url: www.invalid.com
"
`;
